otherlibs/systhreads/thread.mli

```diff
@@ -42,10 +42,21 @@ val id : t -> int
    It can be used to build data structures indexed by threads. *)
 
 val exit : unit -> unit
-(** Terminate prematurely the currently executing thread. *)
+  [@@ocaml.deprecated "Use a dedicated exception."]
+(** @deprecated In order to guarantee the correct reclamation of
```

Comment:
Not sure how all this gets rendered by the compiler warning. But I think the warning should indicate that the semantics was changed and in which version it was changed. Otherwise it seems an innocuous change.

Reply:
This is rendered as a deprecation warning, followed by the message "Use a dedicated exception."

I followed the general pattern for deprecating functions in the standard library: a short message at the compilation, and some more complete explanations in the ocamldoc documentation.

Reply:
(I was unsure because I never remember if @deprecated tags messages get lifted into 
`@@ocaml.deprecated`). Then I think the message should really mention about the semantic change, so that it prompts a careful review. The current message can easily be dismissed as  "ok, something that will be removed in the future, nothing to care about".

Reply:
Better now?

Reply:
Yes but rather than `now`, please specify `Since XX.YY.ZZ, Thread.exit raises an exception`. It would be nice to also document in which version the semantic change happened in the doc string.

Reply:
Done.